cmd/ocfclient/main.go

```diff
@@ -0,0 +1,546 @@
+		}
+	}
+
+	if opts.IdentityTrustCAKey != "" {
+		identityTrustCAKey, err := ioutil.ReadFile(opts.IdentityTrustCAKey)
+		if err != nil {
+			fmt.Println("Reading IdentityTrustCAKey was failed : " + err.Error())
+		} else {
+			fmt.Println("Reading IdentityTrustCAKey from file (" + opts.IdentityTrustCAKey + ") was successful")
+			IdentityTrustedCAKey = identityTrustCAKey
+		}
+	}
+}
+
+func NewSDKClient() (*local.Client, error) {
+	mfgTrustedCABlock, _ := pem.Decode(MfgTrustedCA)
+	if mfgTrustedCABlock == nil {
+		return nil, fmt.Errorf("mfgTrustedCABlock is empty")
+	}
+	mfgCA, err := x509.ParseCertificates(mfgTrustedCABlock.Bytes)
+	if err != nil {
+		return nil, err
+	}
+	mfgCert, err := tls.X509KeyPair(MfgCert, MfgKey)
+	if err != nil {
+		return nil, fmt.Errorf("cannot X509KeyPair: %w", err)
+	}
+
+	identityTrustedCABlock, _ := pem.Decode(IdentityTrustedCA)
+	if identityTrustedCABlock == nil {
+		return nil, fmt.Errorf("identityTrustedCABlock is empty")
+	}
+	identityTrustedCACert, err := x509.ParseCertificates(identityTrustedCABlock.Bytes)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse cert: %w", err)
+	}
+
+	cfg := local.Config{
+		DisablePeerTCPSignalMessageCSMs: true,
+		DeviceOwnershipSDK: &local.DeviceOwnershipSDKConfig{
+			ID:      CertIdentity,
+			Cert:    string(IdentityIntermediateCA),
+			CertKey: string(IdentityIntermediateCAKey),
+		},
+	}
+
+	client, err := local.NewClientFromConfig(&cfg, &SetupSecureClient{
+		mfgCA:   mfgCA,
+		mfgCert: mfgCert,
+		ca:      append(identityTrustedCACert),
+	}, test.NewIdentityCertificateSigner, func(err error) { fmt.Print(err) },
+	)
+	if err != nil {
+		return nil, err
+	}
+	err = client.Initialization(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	return client, nil
+}
+
+func NewClient() *local.Client {
+	appCallback, err := app.NewApp(nil)
+	if err != nil {
+		panic(err)
+	}
+	c, err := local.NewClientFromConfig(&local.Config{
+		KeepAliveConnectionTimeoutSeconds: 3600,
+		ObserverPollingIntervalSeconds:    10,
+	}, appCallback, test.NewIdentityCertificateSigner, func(error) {})
+	if err != nil {
+		panic(err)
+	}
+	return c
+}
+
+func NewSecureClient() (*local.Client, error) {
+	mfgTrustedCABlock, _ := pem.Decode(MfgTrustedCA)
+	if mfgTrustedCABlock == nil {
+		return nil, fmt.Errorf("mfgTrustedCABlock is empty")
+	}
+	mfgCA, err := x509.ParseCertificates(mfgTrustedCABlock.Bytes)
+	if err != nil {
+		return nil, err
+	}
+	mfgCert, err := tls.X509KeyPair(MfgCert, MfgKey)
+	if err != nil {
+		return nil, fmt.Errorf("cannot X509KeyPair: %w", err)
+	}
+	cfg := local.Config{
+		DeviceOwnershipSDK: &local.DeviceOwnershipSDKConfig{
+			ID:      CertIdentity,
+			Cert:    string(IdentityIntermediateCA),
+			CertKey: string(IdentityIntermediateCAKey),
+		},
+	}
+
+	client, err := local.NewClientFromConfig(&cfg, &SetupSecureClient{
+		mfgCA:   mfgCA,
+		mfgCert: mfgCert,
+	}, test.NewIdentityCertificateSigner, func(err error) { fmt.Print(err) },
```

Comment:
the arg `func(err error) { fmt.Print(err) }` it's just for debug purpose so you can change it to `func(err error) {}`